### DISCUSSION TITLE
Is there solution for the case that agent and collect_server are network isolated

### DISCUSSION THREAD

**Discussion by SimonLiuhd** [apache/skywalking#7627] [Q&A]
Hi all,

    The application with java agent deployed in customer's environment, which is separated with internet, while the collector back-end server is deployed in my company network.
    The challenge are followings:
    1> the java agent writes the tracing information collected on the local disk system (such as a local file)
    2> the collector back-end service  imports the above offline tracing files, and shows the tracing chains on UI
    Is the above possible? Or is there another ideal solution for the isolated environment case?

Any suggestion is appreciated. Many thanks in advance.

Simon

**Comment by wu-sheng** [comment]
> Is the above possible? Or is there another ideal solution for the isolated environment case?

Possible, yes, but the community don't have this. This is too special case to everyone. And SkyWalking is a monitoring system, rather than a BI system. This is not our use case.

**Reply by SimonLiuhd** [reply]
Thank your so fast reply, Sheng.

I think what you mean by "Possible" is that we can extend the java-agent and collector server by coding to achieve it, right?
Could you please give me some key points or references about that?

Simon

**Reply by wu-sheng** [reply]
This would not be easy. Take a look how Kafka reporter works, change it to file reporter. 
But after that, we still have to deal with some gRPC not connected event in the core. `GRPCChannelManager` has to be set in `connected` status.(No recommend way, because we never expect to change this.)

**Reply by SimonLiuhd** [reply]
Thanks a lot. Great helps for me.

Simon
